File: src/apps/Andriod/app/src/main/java/com/dkellycollins/triggerfinger/managers/entity/IWeaponEntityManager.java
package com.dkellycollins.triggerfinger.managers.entity;

import com.dkellycollins.triggerfinger.data.entity.IWeapon;

public interface IWeaponEntityManager {

    Iterable<IWeapon> retrieve();
    IWeapon retrieve(int id);

    int create(int collidableId);

    void delete(int id);

}

File: src/apps/Android-Kotlin/app/src/main/kotlin/com/dkellycollins/triggerfinger/managers/entity/IWeaponEntityManager.kt
package com.dkellycollins.triggerfinger.managers.entity

import com.dkellycollins.triggerfinger.data.entity.IWeapon

interface IWeaponEntityManager {

    fun retrieve(): Iterable<IWeapon>
    fun retrieve(id: Int): IWeapon

    fun create(collidableId: Int): Int

    fun delete(id: Int)

}
